core/src/apps/binance/helpers.py

```diff
@@ -0,0 +1,115 @@
+from micropython import const
+
+from trezor.crypto import bech32
+from trezor.crypto.scripts import sha256_ripemd160_digest
+from trezor.messages.BinanceCancelMsg import BinanceCancelMsg
+from trezor.messages.BinanceInputOutput import BinanceInputOutput
+from trezor.messages.BinanceOrderMsg import BinanceOrderMsg
+from trezor.messages.BinanceSignTx import BinanceSignTx
+from trezor.messages.BinanceTransferMsg import BinanceTransferMsg
+
+from apps.common import HARDENED
+
+ENVELOPE_BLUEPRINT = '{{"account_number":"{account_number}","chain_id":"{chain_id}","data":null,"memo":"{memo}","msgs":[{msgs}],"sequence":"{sequence}","source":"{source}"}}'
+MSG_TRANSFER_BLUEPRINT = '{{"inputs":[{inputs}],"outputs":[{outputs}]}}'
+MSG_NEWORDER_BLUEPRINT = '{{"id":"{id}","ordertype":{ordertype},"price":{price},"quantity":{quantity},"sender":"{sender}","side":{side},"symbol":"{symbol}","timeinforce":{timeinforce}}}'
+MSG_CANCEL_BLUEPRINT = '{{"refid":"{refid}","sender":"{sender}","symbol":"{symbol}"}}'
+INPUT_OUTPUT_BLUEPRINT = '{{"address":"{address}","coins":[{coins}]}}'
+COIN_BLUEPRINT = '{{"amount":"{amount}","denom":"{denom}"}}'
+
+# 1*10^18 Jagers equal 1 BNB https://www.binance.vision/glossary/jager
+DIVISIBILITY = const(18)
+
+
+def produce_json_for_signing(envelope: BinanceSignTx, msg) -> str:
+    if isinstance(msg, BinanceTransferMsg):
+        json_msg = produce_transfer_json(msg)
+    elif isinstance(msg, BinanceOrderMsg):
+        json_msg = produce_neworder_json(msg)
+    elif isinstance(msg, BinanceCancelMsg):
+        json_msg = produce_cancel_json(msg)
+    else:
+        raise ValueError("input message unrecognized, is of type " + type(msg).__name__)
+
+    if envelope.source is None or envelope.source <= 0:
```

Comment:
is the `<=` correct? is 0 an invalid value?

Reply:
"set to zero if unwilling to disclose." so you are right, it should be <, not <=. will fix